presto-main/src/test/java/io/prestosql/operator/scalar/AbstractTestExtract.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.operator.scalar;
+
+import com.google.common.base.CaseFormat;
+import io.prestosql.spi.PrestoException;
+import io.prestosql.sql.query.QueryAssertions;
+import io.prestosql.sql.tree.Extract;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public abstract class AbstractTestExtract
+{
+    protected QueryAssertions assertions;
+
+    @BeforeClass
+    public void init()
+    {
+        assertions = new QueryAssertions();
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+    {
+        assertions.close();
+        assertions = null;
+    }
+
+    @Test
+    public void testTestCompleteness()
+            throws NoSuchMethodException
+    {
+        for (Extract.Field value : Extract.Field.values()) {
+            String testMethodName = "test" + CaseFormat.UPPER_UNDERSCORE.to(CaseFormat.UPPER_CAMEL, value.name());
+            this.getClass().getMethod(testMethodName);
```

Comment:
Catch and throw an error with a better message

Reply:
This currently looks like this:
```
java.lang.NoSuchMethodException: io.prestosql.operator.scalar.time.TestExtract.testQuarter()

	at java.base/java.lang.Class.getMethod(Class.java:2108)
	at io.prestosql.operator.scalar.AbstractTestExtract.testTestCompleteness(AbstractTestExtract.java:52)
```

i felt this is sufficient?

Reply:
Ok, that's fine.